### ISSUE TITLE
Test Case fales to run

### ISSUE THREAD

**Issue by zoobar** [squoilin/Dispa-SET#1]
Whenever I try to solve the test case with Pyomo. The process stops with the following error message:

KeyError: "Error setting parameter value: Index '[6] - WindOn_Z2' is not valid for array Param 'StorageCapacity'"

I investigated the dictionary in SimData but everything looks fine and consistent. No, idea why Pyomo struggles with this line. All other cases also struggle with a similar error message for 'StorageCapacity'. Many thanks for your help.

**Comment by squoilin** [comment]
Unfortunately, we didn't have time to keep on with the updating of the Pyomo version of Dispa-SET.

This runs fine in GAMS, but the Pyomo code need to be updated.  Of course any help is always welcome!

**Comment by matteodefelice** [comment]
I am trying to find the issue here, for example in `model.py:101` we have:
```
    model.LoadShedding = Param(sets['n'], initialize=params['LoadShedding'])  # [n.a.] Load shedding capacity
```
but according to the model description `LoadShedding` should be `LoadShedding(n,h)`. However, if I change the line to add `sets[h]` I get another error, this time when setting the `StorageCapacity` parameter:
```
ERROR: Constructing component 'StorageCapacity' from data=None failed:
	KeyError: "Error setting parameter value: Index '[6] - WindOn_Z2' is not valid for array Param 'StorageCapacity'"
```

**Comment by etesio** [comment]
Hi @matteodefelice, just jumping in as I also looked a bit in the pyomo version some time ago.. I think the point here is that you need the set 'u' set in defining the StorageCapacity parameter. This should also reverberate in defining some constraints down the line (LoadShedding and StorageBoundaries, I believe).

**Comment by matteodefelice** [comment]
Exactly, I am trying to isolate the errors here, it seems that both `EQ_LoadShedding` and `EQ_Storage_boundaries` contain an error. For the latter, I am trying to inspect if `StorageFinalMin` could be fixed in some way, it seems the source of error in the constraint rule.
